### ISSUE TITLE
[Idea] Progress bar for any operation that takes some longer time

### ISSUE THREAD

**Issue by sfuchs79** [subsurface/subsurface#373]
Create a standard progress bar pop up window or integrate a progress bar in the main window.

Show/activate this progress bar for any operation that takes a longer time. Up to now I mainly identified any image processing tasks as candidates but maybe there are other things.

**Comment by mturkia** [comment]
Also import and export functions might take quite long. Unfortunately the XSLT transformation used for this does not really support reporting of progress... So version of the progress bar that just indicates something happening in this case would be in order.

**Comment by dirkhh** [comment]
Yeah, having a busy indicator at least. I tend to have a reasonably fast machine, so I haven't seen anything that takes unreasonably long... can we collect specific actions that are good candidates for this? For example, exactly which image operations should have this? I think it would be reasonable to do this for all import and export operations that affect more than 5 dives or something. Or maybe just for all of them, but with a one second delay before it is shown. I need to re-read the code that I did for the cloud operations but I seem to remember there was something like this (but of course the other problem is one of not giving cycles to the main UI... I remember that that was a big issue with getting a decent progress report to work with the cloud operations...)

**Comment by neolit123** [comment]
+1 for collecting a list of actions. (oops closed by mistake)

**Comment by neolit123** [comment]
so the progress bar in the `MainWindow` can be potentially re-used for this.

for back-end work something like `QtConcurrency` is the correct choice, which creates a new thread from the thread pool and is flexible enough so that one does not need to subclass `QThread`.
but if one needs to update UI elements, the signals and slots mechanic needs to be used, so that the main thread can receive updates. i suspect that we need to update UI elements in some of the work that needs a progress bar so that complicates things.

@sfuchs79 
can you provide a quick list of operations that need a progress bar?
